test/ng/directive/ngEventDirsSpec.js

```diff
@@ -148,6 +148,163 @@ describe('event directives', function() {
       expect($rootScope.blur).toHaveBeenCalledOnce();
       expect(element.val()).toBe('newValue');
     }));
+  });
+
+  describe('events in digest phases', function() {
+
+    it('should not stop the digest when the handler throws an error', function() {
+
+      module(function($exceptionHandlerProvider) {
+        $exceptionHandlerProvider.mode('log');
+      });
+
+      inject(function($rootScope, $compile, $exceptionHandler, $log) {
+
+        element = $compile('<button ng-click="click()">Click</button>')($rootScope);
+        expect($log.assertEmpty());
+        $rootScope.click = function() {
+          throw new Error('listener error');
+        };
+
+        $rootScope.do = function() {
+          element.triggerHandler('click');
+          $log.log('done');
+        };
+
+        $rootScope.$apply(function() {
+          $rootScope.do();
+        });
+
+        expect($exceptionHandler.errors).toEqual([Error('listener error')]);
+        expect($log.log.logs).toEqual([['done']]);
+        $log.reset();
+      });
+    });
+
+
+    it('should not stop the digest when the handler throws an error', function() {
+
+      module(function($exceptionHandlerProvider) {
+        $exceptionHandlerProvider.mode('log');
+      });
+
+      inject(function($rootScope, $compile, $exceptionHandler, $log) {
+
+        element = $compile('<button ng-click="click()">Click</button>')($rootScope);
+        expect($log.assertEmpty());
+        $rootScope.click = function() {
+          throw new Error('listener error');
+        };
+
+        $rootScope.do = function() {
+          element.triggerHandler('click');
+          $log.log('done');
+        };
+
+        $rootScope.do();
+
+        expect($exceptionHandler.errors).toEqual([Error('listener error')]);
+        expect($log.log.logs).toEqual([['done']]);
+        $log.reset();
+      });
+    });
+
+    it('should not stop the digest when the handler is called in a watch expression fn', function() {
```

Comment:
How is this test (and the ones below) test that the digest is not stopped? I would expect a second watch expression.
(What am I missing?)